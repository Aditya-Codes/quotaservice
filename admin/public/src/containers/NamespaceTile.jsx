import PropTypes from 'prop-types';
import React from 'react';
import { Component } from 'react';

export default class NamespaceTile extends Component {
  onClick = () => {
    const { namespace, selectNamespace } = this.props
    selectNamespace(namespace.name)
  }

  render() {
    const { isSelected, namespace } = this.props
    const buckets = namespace.buckets || {}

    return (<div className={'flex-box flex-tile namespace' + (isSelected ? ' selected' : '')} onClick={this.onClick}>
      <p className="title">{namespace.name}</p>
      <hr />
      {this.renderBucket(namespace.dynamic_bucket_template)}
      {this.renderBucket(namespace.default_bucket)}
      {Object.keys(buckets).map(key =>
        this.renderBucket(buckets[key])
      )}
    </div>)
  }

  renderBucket(bucket) {
    if (!bucket)
      return

    return (<div key={bucket.name} className="bucket">{bucket.name}</div>)
  }
}

NamespaceTile.propTypes = {
  isSelected: PropTypes.bool.isRequired,
  namespace: PropTypes.object.isRequired,
  selectNamespace: PropTypes.func.isRequired
}
