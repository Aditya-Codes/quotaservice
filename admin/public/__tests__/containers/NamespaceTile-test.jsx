import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';

import NamespaceTile from '../../src/containers/NamespaceTile.jsx';

describe('NamespaceTile', () => {
  it('renders with minimum data', () => {
    const props = {
      namespace: {},
      selectNamespace: () => null,
      isSelected: false,
    };
    const component = shallow(<NamespaceTile {...props} />);
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
  });

  it('renderes in selected state', () => {
    const props = {
      namespace: {},
      selectNamespace: () => null,
      isSelected: true,
    };
    const component = shallow(<NamespaceTile {...props} />);
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
    expect(component.find('.namespace.selected')).toHaveLength(1);
  });
});
