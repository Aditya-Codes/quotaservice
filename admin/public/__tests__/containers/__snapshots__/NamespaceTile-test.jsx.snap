// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamespaceTile renderes in selected state 1`] = `
<div
  className="flex-box flex-tile namespace selected"
  onClick={[Function]}
>
  <p
    className="title"
  />
  <hr />
</div>
`;

exports[`NamespaceTile renders with minimum data 1`] = `
<div
  className="flex-box flex-tile namespace"
  onClick={[Function]}
>
  <p
    className="title"
  />
  <hr />
</div>
`;
